test/e2e/tests/confirmations/signatures/sign-typed-data-v4.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { strict as assert } from 'assert';
+import { Suite } from 'mocha';
+import { withRedesignConfirmationFixtures } from '../helper-fixture';
+import {
+  DAPP_HOST_ADDRESS,
+  WINDOW_TITLES,
+  openDapp,
+  unlockWallet,
+} from '../../../helpers';
+import { Ganache } from '../../../seeder/ganache';
+import { Driver } from '../../../webdriver/driver';
+
+describe('Confirmation Signature - Sign Typed Data V4', function (this: Suite) {
+  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) { return; }
+
+  it('initiates and confirms', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver, ganacheServer }: { driver: Driver, ganacheServer: Ganache }) => {
+        const addresses = await ganacheServer.getAccounts();
+        const publicAddress = addresses?.[0] as string;
+
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await driver.clickElement('#signTypedDataV4');
```

Comment:
Is everything up to this line the same in every test in every file?

Should we create a `createSignatureConfirmation` function in that helper file I mentioned in the other comment?

It could accept an enum such as `SignatureType` to encapsulate the button ID we need to click also.

Reply:
I like this idea. Could be used outside of the confirmation redesign files as well. Created an issue ticket to handle this after the release https://github.com/MetaMask/metamask-extension/issues/24468